### ISSUE TITLE
Working with auth.shop middleware

### ISSUE THREAD

**Issue by blackdogdevelopers** [ohmybrew/laravel-shopify#323]
Howdy,

In addition to being kind of dumb, I'm kind of new to all of this and am confused about how the auth.shop middleware works.

I have setup a basic app and a single page is working.
However if I try to use the middleware on a second route it fails with unable to verify signature exception.

For example. Getting the default route / works, but not the test
```php
Route::get('/', function () {
    return view('app');
})->middleware(['auth.shop'])->name('home');

Route::get('/test', function () {
    return view('app');
})->middleware(['auth.shop'])->name('test');
```

In the app.blade.php I have this 

```php
<a href="{{ shop_route('test') }}">Test route</a>
```
It correctly generates the URL with the shop address in it. https://blah.com/?shop=blorp.com
However, when trying to hit it I get the exception.

Are you able to provide examples of how to correctly build routes that can pass auth.shop?

Ideally I want to be able to use it against axios.get() calls to hit an api in the application.

I've noticed that there is some issues with ngrok (which I'm using) but not sure if they are relevant to this.

D.

**Comment by StefanNeuser** [comment]
Howdy, i guess there are missing the params for the hmac-verification. Take a look into AuthShop.php

**Comment by ronald2wing** [comment]
I also have this exception:  production.ERROR: Unable to verify signature. {"exception":"[object] (Exception(code: 0): Unable to verify signature.
The url generated by the shop_route returns a 500 error page.

**Comment by blackdogdevelopers** [comment]
@StefanNeuser thanks for the info, it does appear to just be missing data in the URL (hmac & timestamp). I figured that I didn't actually need it in my particular instance and just removed the auth.shop middleware from the route.

I have found a call which can apparently generate a hmac token [ShopifyApp::createHmac()], but I didn't dig deep enough to figure out what the $opts input parameter options were to get it working.
